### PR TITLE
Add more metadata

### PR DESCRIPTION
**Pull request by kcudnik**
- Add missing _START and _END to attributes list
- Fix object type naming
- Add more sanity checks

### CODE REVIEW

File: inc/saitypes.h [hunk 1]
```diff
@@ -157,51 +157,51 @@ typedef enum _sai_common_api_t {
...
+    SAI_OBJECT_TYPE_HOSTIF_USER_DEFINED_TRAP = 34,
+    SAI_OBJECT_TYPE_NEIGHBOR_ENTRY           = 35,
+    SAI_OBJECT_TYPE_ROUTE_ENTRY              = 36,
+    SAI_OBJECT_TYPE_VLAN                     = 37,
+    SAI_OBJECT_TYPE_VLAN_MEMBER              = 38,
+    SAI_OBJECT_TYPE_HOSTIF_PACKET            = 39,
+    SAI_OBJECT_TYPE_TUNNEL_MAP               = 40,
+    SAI_OBJECT_TYPE_TUNNEL                   = 41,
+    SAI_OBJECT_TYPE_TUNNEL_TERM_TABLE_ENTRY  = 42,
+    SAI_OBJECT_TYPE_FDB_FLUSH                = 43,
```
Comment (lguohan) [inline inc/saitypes.h]: is this an object now?

File: meta/saisanitycheck.c [hunk 2]
```diff
@@ -1216,6 +1221,79 @@ void check_single_object_type_attributes(
...
+
+        int last = -1;
+
+        /* check all listed attributes under this object type */
+
+        for (; meta[index] != NULL; ++index)
+        {
+            const sai_attr_metadata_t* am = meta[index];
+
+            META_ASSERT_TRUE((int)am->attrid >= 0, "attrmust be non negative");
```
Comment (lguohan) [inline meta/saisanitycheck.c]: attribute must be non negative

File: inc/saitypes.h (hunk 1 above)
Comment (kcudnik) [inline inc/saitypes.h]: it's not really an object, but will be used to serialize fdb_flush in the future, same usecase as in hostif packet for recv/send

File: meta/saisanitycheck.c (hunk 2 above)
Comment (kcudnik) [inline meta/saisanitycheck.c]: fixed

File: inc/saitypes.h [hunk 3]
```diff
@@ -157,51 +157,51 @@ typedef enum _sai_common_api_t {
...
-    SAI_OBJECT_TYPE_ROUTE_ENTRY             = 36,
-    SAI_OBJECT_TYPE_VLAN                    = 37,
-    SAI_OBJECT_TYPE_VLAN_MEMBER             = 38,
-    SAI_OBJECT_TYPE_PACKET                  = 39,
-    SAI_OBJECT_TYPE_TUNNEL_MAP              = 40,
-    SAI_OBJECT_TYPE_TUNNEL                  = 41,
-    SAI_OBJECT_TYPE_TUNNEL_TERM_TABLE_ENTRY = 42,
-    SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP  = 43,
-    SAI_OBJECT_TYPE_MAX                     = 44
+    SAI_OBJECT_TYPE_NULL                     =  0, /**< invalid object type */
```
Comment (stcheng) [inline inc/saitypes.h]: /**< ? or /*

Comment (kcudnik) [inline inc/saitypes.h]: "<" is doxygen meta tag

**Pull request merged**